validation/metadata_test.go

```diff
@@ -0,0 +1,262 @@
+package validation
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func createObjWithUID() *unstructured.Unstructured {
```

Comment:
`createObj` seems like it could be a function that accepts functional options like `WithUID`, `WithGeneration`, etc. That would reduce a bunch of boilerplate duplication.